src/client/oicp/OICPClient.ts

```diff
@@ -0,0 +1,127 @@
+import AxiosFactory from '../../utils/AxiosFactory';
+import { AxiosInstance } from 'axios';
+import BackendError from '../../exception/BackendError';
+import { HTTPError } from '../../types/HTTPError';
+import OICPEndpoint from '../../types/oicp/OICPEndpoint';
+import { OICPRole } from '../../types/oicp/OICPRole';
+import { OicpSetting } from '../../types/Setting';
+import { ServerAction } from '../../types/Server';
+import Tenant from '../../types/Tenant';
+import { OICPJobResult } from '../../types/oicp/OICPJobResult';
+import { OICPOperatorID } from '../../types/oicp/OICPEvse';
+import Logging from '../../utils/Logging';
+
+const MODULE_NAME = 'OICPClient';
+
+export default abstract class OICPClient {
+  protected axiosInstance: AxiosInstance;
+  protected oicpEndpoint: OICPEndpoint;
+  protected tenant: Tenant;
+  protected role: string;
+  protected settings: OicpSetting;
+
+  protected constructor(tenant: Tenant, settings: OicpSetting, oicpEndpoint: OICPEndpoint, role: string) {
+    if (role !== OICPRole.CPO && role !== OICPRole.EMSP) {
+      throw new BackendError({
+        message: `Invalid OICP role '${role}'`,
+        module: MODULE_NAME, method: 'constructor',
+      });
+    }
+    this.axiosInstance = AxiosFactory.getAxiosInstance(tenant.id);
```

Comment:
Here you have the axios client, just use it in all clients.

Reply:
okay, I changed it